apps/reaper/test/unit/reaper/decoder/json_test.exs

```diff
@@ -63,6 +63,24 @@ defmodule Reaper.Decoder.JsonTest do
       assert result == expected
     end
 
+    data_test "json decoder handles empty arrays with topLevelSelector: #{selector}" do
+      dataset_with_selector = TDG.create_dataset(%{technical: %{sourceFormat: "json", topLevelSelector: selector}})
+
+      body = body |> Jason.encode!()
+      File.write!(@filename, body)
+
+      {:ok, result} = Decoder.Json.decode({:file, @filename}, dataset_with_selector)
+
+      assert result == []
+
+      where([
+        [:body, :selector],
+        [%{data: []}, "$.data[*]"],
+        [%{data: %{nested_data: []}}, "$.data.nested_data[*]"],
+        [%{data: [%{nested_data: []}]}, "$.data[*].nested_data"]
```

Comment:
I'd be interested to know what the positive case for this kind of nesting is, but I "think" that even then, a single empty list works for that

Reply:
It "might?" need to be tweaked to be `$.data[*].nested_data[*]` to be more accurate as to the intent, but once again, not sure if it really matters